### ISSUE TITLE
Both events(componentDidUpdate and componentDidMount) getting fired on page refresh/on page reload  

### ISSUE THREAD

**Issue by spo365** [SharePoint/sp-dev-docs#5809]
## Category

- [ ] Question
We have search box webpart using query string as below :https://test1.sharepoint.com/sites/DE1/SitePages/Search-Results-Page.aspx?q=test&chkBoxStatus=false
when it redirect to searchresult page Both events(componentDidUpdate and componentDidMount) getting fired on page refresh/on page reload. however it is happening intermittent only.    
 
component did mount -----
 
<div>    public async componentDidMount() {

                       let initialCheckBoxValue:boolean;
                        var urlParams:any = new UrlQueryParameterCollection(window.location.href);
                        /* query string parameter chkBoxStatus used to handle title or content based search
                        chkBoxStatus = true means search is based on title only
                        chkBoxStatus = false means search is based on content */
                        if(urlParams.getValue("chkBoxStatus".toUpperCase()))
                        {
                          initialCheckBoxValue = JSON.parse(decodeURIComponent(urlParams.getValue('chkBoxStatus'.toUpperCase())));
                          if(initialCheckBoxValue)
                          {
                            this.props.searchService.queryTemplate = "title:{searchTerms}";
                          }
                          else{
                                this.props.searchService.queryTemplate = "{searchTerms}";
                          }
                        }
                        this.setState({
                            results: searchResults,
                            areResultsLoading: false
                        });

    }</div>

 
<div> public async componentDidUpdate(prevProps: ISearchResultsContainerProps) {
     
            if(this.state.searchCheckBoxStatus ){
                console.log("if");
                this.props.searchService.queryTemplate = "title:{searchTerms}";
            }
            else {
                console.log("else");
                this.props.searchService.queryTemplate = "{searchTerms}";
            }
}
  --- Issue ---</div>

Both events getting fired on page refresh/on page reload 

## Environment details (*development & target environment*)

> **DELETE THIS LINE BEFORE SUBMITTING** - Complete the following (*if ignored you'll be prompted for it before we can address your issue... save the time and provide it in your initial submission*).

- **OS**: [e.g. Windows 10 | MacOS 10.15.x]
- **Target Environment**: [| SharePoint Online]
- **Framework**: [SPFX]
- **Browser(s)**: [e.g. Chrome v79 | Safari v22 | FireFox v31 | IE v11 | Edge | Chromium Edge v79]
- **Tooling**: [e.g. VS Code | SPFx v1.10.0 | Visual Studio 2019]
- **Additional details**: *The more context you can provide, the easier it is (and therefore quicker) to help.*

## Submission guidelines (delete this section before submitting)

*Thanks for your contribution! Sharing is caring.*

**Comment by patmill** [comment]
Those lifecycle events can fire at various times, and may change over time, depending on the app, caching, navigation, etc. etc.  If there is a specific issue, please feel free to open a new issue w/ the error case, etc.
